rope/base/utils.py

```diff
@@ -81,3 +81,48 @@ def __call__(self, *args, **kwds):
         if len(self.cache) > self.count:
             del self.cache[0]
         return result
+
+class memoize(object):
```

Comment:
I have rebased your branch on top of my https://github.com/mcepl/rope/tree/sublimeText_speedup and squashed it to one commit (and pushed to my repo as https://github.com/mcepl/rope/tree/cache_optimization) and created new pull request to https://github.com/mcepl/rope/tree/sublimeText_speedup as https://github.com/mcepl/rope/pull/1

Unfortunately, it still fails in CI: https://travis-ci.org/mcepl/rope/builds/111960979 ... some syntax (dict generators) is certainly not ready for py26, and most likely I have screwed up something when rebasing. Please, comments on https://github.com/mcepl/rope/pull/1